src/helper/UserSession.cpp

```diff
@@ -50,11 +51,37 @@ namespace SDDM {
 
     bool UserSession::start() {
         auto helper = qobject_cast<HelperApp*>(parent());
-        QProcessEnvironment env = helper->session()->processEnvironment();
+        QProcessEnvironment env = processEnvironment();
 
         bool isWaylandGreeter = false;
         if (env.value(QStringLiteral("XDG_SESSION_TYPE")) == QLatin1String("x11")) {
             QString command;
+            // Create the Xauthority file
+            QByteArray cookie = qobject_cast<HelperApp*>(parent())->cookie();
+            if (cookie.isEmpty())
+                return false;
+
+            // Place it into /tmp, which is guaranteed to be read/writeable by
+            // everyone while having the sticky bit set to avoid messing with
+            // other's files.
+            m_xauthFile.setFileTemplate(QStringLiteral("/tmp/xauth_XXXXXX"));
+
+            if (!m_xauthFile.open()) {
+                qCritical() << "Could not create the Xauthority file";
+                return false;
+            }
+
+            QString display = processEnvironment().value(QStringLiteral("DISPLAY"));
+            qDebug() << "Adding cookie to" << m_xauthFile.fileName();
+
+            if (!XAuth::addCookieToFile(m_xauthFile.fileName(), display, cookie)) {
```

Comment:
Wrong order of parameters:

```suggestion
            if (!XAuth::addCookieToFile(display, m_xauthFile.fileName(), cookie)) {
```